okta-angular.js

```diff
@@ -0,0 +1,339 @@
+				sessionToken: options.token,
+				responseType : options.responseType,
+				scope : options.scopes
+			}).then(function(res) {
+				var tokens = {};
+				if(angular.isArray(res)){
+					angular.forEach(res, function(val) {
+						if("accessToken" in val){
+							tokens["accessToken"] = val;
+						}
+						if ("idToken" in val){
+							tokens["idToken"] = val;
+						}
+					});
+				} else {
+					if (res.hasOwnProperty("accessToken")){ tokens["accessToken"] = res; }
+					if (res.hasOwnProperty("idToken")){ tokens["idToken"] = res; }
+				}
+				deferred.resolve(tokens);
+			});
+		}
+		return deferred.promise;
+	}
+
+	/**
+	 *	Refreshes the current session
+	 */
+	var refreshSession = function() {
+		var deferred = $q.defer();
+		auth.session.exists()
+		.then(function(res) {
+			if(res == true) {
+				auth.session.refresh()
+				.then(function(success){
+					deferred.resolve(success);
+				});
+			}
+		})
+		.fail(function(err){deferred.reject(err);});
+		return deferred.promise;
+	}
+
+	/**
+	 * 	Closes the current session
+	 */
+	var closeSession = function() {
+		var deferred = $q.defer();
+		auth.session.close()
+		.finally(function(){
+			deferred.resolve("Closed Session");
+		})
+		.fail(function(err){deferred.reject(err);});
+		return deferred.promise;
+	}
+
+	/**
+	 *	Renews the current ID token
+	 */
+	var renewIdToken = function(options) {
+		var scopes = {'scope' : options};
+		var deferred = $q.defer();
+		auth.idToken.refresh(scopes)
+		.then(function(res) {
+			deferred.resolve({
+				"idToken" : res
+			});
+		})
+		.fail(function(err){deferred.reject(err);});
+		return deferred.promise;
+	}
+
+	/**
+	 *	Given an "idToken" or "accessToken", it decodes the
+	 *	header, payload, and signiture
+	 */
+	var decodeIdToken = function(token) {
+		var deferred = $q.defer();
+		var decoded = auth.idToken.decode(token);
+		if (!angular.isUndefined(decoded)){
+			deferred.resolve(decoded);
+		} else {
+			deferred.reject(token);
+		}
+		return deferred.promise;
+	}
+
+	/**
+	 *	Logs the user out of the current session
+	 */
+	var signout = function() {
+		var deferred = $q.defer();
+		auth.session.exists()
+		.then(function(exists){
+			if(exists) {
+				auth.signOut();
+			}
+			deferred.resolve("Signed out");
+		})
+		.fail(function(err){
+			deferred.reject(err);
+		});
+		return deferred.promise;
+	}
+
+	/**
+	 *	Return functions
+	 */
+	return {
+		create : create, 
+		login : login,
+		getTokens : getTokens,
+		refreshSession : refreshSession,
+		closeSession : closeSession,
+		renewIdToken : renewIdToken,
+		decodeIdToken : decodeIdToken,
+		signout : signout
+	}
+})
+
+/********************************************************/
+/********************************************************/
+
+/********************************************************/
+/********************************************************/
+
+/**
+ *	Wrapper for Sign-In Widget
+ */	
+.factory("widgetClient", function($q) {
+	var auth;
```

Comment:
Same comment for this - why are we introducing a new interface here? It looks like it's doing what we're already doing in the sign-in-widget, but using different method names with different response formats.